dashboard/test/controllers/activities_controller_test.rb

```diff
@@ -1343,4 +1341,15 @@ def new_level
     post :milestone, params: milestone_params
     assert_response 403
   end
+
+  test "milestone strips emoji from program and saves it" do
+    params = @milestone_params
+    params[:program] = panda_panda
+
+    post :milestone, params: params
+
+    user_level = UserLevel.last
+    # panda_panda contains a panda emoji, ensure that it's gone
+    assert_equal user_level.level_source.data, 'Panda'
```

Comment:
It comes from [here](https://github.com/code-dot-org/code-dot-org/blob/3bbf9925948ebf5399bbf0064c000ff3e09cf521/dashboard/test/test_helper.rb#L76), namely the characters in `panda_panda` that are not stripped.